kubetest2-gke/deployer/up.go

```diff
@@ -67,22 +67,26 @@ func (d *deployer) Up() error {
 	loc := location(d.region, d.zone)
 	for i := range d.projects {
 		project := d.projects[i]
-		subNetworkArgs := subNetworkArgs(d.projects, d.region, d.network, i)
+		subNetworkArgs := subNetworkArgs(d.autopilot, d.projects, d.region, d.network, i)
 		for j := range d.projectClustersLayout[project] {
 			cluster := d.projectClustersLayout[project][j]
 			privateClusterArgs := privateClusterArgs(d.projects, d.network, d.privateClusterAccessLevel, d.privateClusterMasterIPRanges, cluster)
 			eg.Go(func() error {
 				// Create the cluster
-				args := make([]string, len(d.createCommand()))
-				copy(args, d.createCommand())
+				args := d.createCommand()
 				args = append(args,
 					"--project="+project,
 					loc,
-					"--machine-type="+d.machineType,
-					"--image-type="+image,
-					"--num-nodes="+strconv.Itoa(d.nodes),
 					"--network="+transformNetworkName(d.projects, d.network),
 				)
+				// A few args are not supported in GKE Autopilot cluster creation, so they should be left unset.
+				// https://cloud.google.com/sdk/gcloud/reference/container/clusters/create-auto
+				if !d.autopilot {
+					args = append(args, "--machine-type="+d.machineType)
+					args = append(args, "--num-nodes="+strconv.Itoa(d.nodes))
+					args = append(args, "--image-type="+d.imageType)
+				}
```

Comment:
It might be worth refactoring all the argument builder logic somewhere as a follow up.